### DISCUSSION TITLE
OnlineDDL Scheduler

### DISCUSSION THREAD

**Discussion by earayu** [wesql/wescale#257] [General]
We talk about OnlineDDL Executor here.

doc: https://github.com/vitessio/vitess/blob/main/doc/design-docs/OnlineDDLScheduler.md

**Comment by earayu** [comment]
Fun Fact:

`engine.OnlineDDL` would add some hints ahead of DDL statement in this function: `NewOnlineDDL(...)`
```
-- Origin DDL Statement 
create table t3 (c1 int primary key);

-- DDL After rewritten
create /*vt+ uuid="34383663343930635f353039325f313165655f396461325f666163383733643365366365" context="7674676174653a33316262663162322d353039322d313165652d396461322d666163383733643365366365" table="7433" strategy="6f6e6c696e65" options="" */ table t3 (
	c1 int primary key
)
```

**Comment by earayu** [comment]
If there are frequent timeouts when debugging vttablet, you can configure the queryserver-config-query-timeout parameter.

**Reply by gerayking** [reply]
got it

**Comment by earayu** [comment]
Check your knowledge of OnlineDDL:
- [ ] The meaning of every column of system table `schema_migrations`

**Comment by earayu** [comment]
# states & transitions

![image](https://github.com/wesql/wesql-scale/assets/13747699/d459ea56-e5f0-4f1e-8ff9-c81d55e98a4c)

**Comment by gerayking** [comment]
For the `onlineDDL.executor` connection pool, why do we create new connections instead of utilizing all the connections available in the connection pool? For example, all sql execute by `execQuery`
